packages/plugin-core/src/features/BacklinksTreeDataProvider.ts

```diff
@@ -6,239 +6,66 @@ import {
- */
-const refsToBacklinkTreeItems = (
-  refs: FoundRefT[],
-  fsPath: string,
-  parent: Backlink
-) => {
-  return refs.map((ref) => {
-    const lineNum = ref.location.range.start.line;
-    const backlink = new Backlink(
-      ref.matchText,
-      undefined,
-      vscode.TreeItemCollapsibleState.None
-    );
-    backlink.parentBacklink = parent;
-    backlink.description = `on line ${lineNum + 1}`;
-    backlink.tooltip = ref.matchText;
-    backlink.command = {
-      command: "vscode.open",
-      arguments: [ref.location.uri, { selection: ref.location.range }],
-      title: "Open File",
-    };
-    if (ref.isCandidate) {
-      backlink.command = {
-        command: "dendron.convertCandidateLink",
-        title: "Convert Candidate Link",
-        arguments: [{ location: ref.location, text: path.parse(fsPath).name }],
-      };
-    }
-    return backlink;
-  });
-};
+import { Backlink, BacklinkFoundRef } from "./Backlink";
 
 export default class BacklinksTreeDataProvider
   implements vscode.TreeDataProvider<Backlink>
 {
+  private _onDidChangeTreeDataEmitter: vscode.EventEmitter<
+    Backlink | undefined | void
+  >;
+  private _engineEvents;
   private _sortOrder: BacklinkSortOrder = BacklinkSortOrder.PathNames;
-  private _onDidChangeTreeData: vscode.EventEmitter<Backlink | undefined> =
-    new vscode.EventEmitter<Backlink | undefined>();
-  readonly onDidChangeTreeData: vscode.Event<Backlink | undefined> =
-    this._onDidChangeTreeData.event;
-  readonly isLinkCandidateEnabled: boolean | undefined;
+  readonly _isLinkCandidateEnabled: boolean | undefined;
+
+  /**
+   * Signals to vscode UI engine that the backlinks view needs to be refreshed.
+   */
+  readonly onDidChangeTreeData: vscode.Event<Backlink | undefined | void>;
+
+  /**
+   *
+   * @param engineEvents - specifies when note state has been changed on the
+   * engine
+   */
+  constructor(
+    engineEvents: EngineEventEmitter,
+    isLinkCandidateEnabled: boolean | undefined
+  ) {
+    this._isLinkCandidateEnabled = isLinkCandidateEnabled;
+    this.updateSortOrder(BacklinkSortOrder.PathNames);
 
-  constructor(isLinkCandidateEnabled: boolean | undefined) {
-    this.isLinkCandidateEnabled = isLinkCandidateEnabled;
+    this._onDidChangeTreeDataEmitter = new vscode.EventEmitter<
+      Backlink | undefined | void
+    >();
 
-    this.updateSortOrder(BacklinkSortOrder.PathNames);
+    this.onDidChangeTreeData = this._onDidChangeTreeDataEmitter.event;
+    this._engineEvents = engineEvents;
+    this.setupSubscriptions();
   }
 
-  refresh(): void {
-    this._onDidChangeTreeData.fire(undefined);
+  private setupSubscriptions(): void {
+    this._engineEvents.onEngineNoteStateChanged(() => {
+      const ctx = "refreshBacklinks";
+      Logger.info({ ctx });
+      this.refreshBacklinks();
+    });
   }
 
+  /**
+   * Tells VSCode to refresh the backlinks view. Debounced to fire every 100 ms
+   */
+  public refreshBacklinks = _.debounce(() => {
```

Comment:
make private

Reply:
Ah gotcha - let's still make it private though and encapsulate the subscription to `onDidChangeActiveTextEditor` within `BacklinksTreeDataProvider`.

So basically move this logic: 

```typescript
    vscode.window.onDidChangeActiveTextEditor(() =>
      backlinksTreeDataProvider.refreshBacklinks()
    );
```

into here:
```typescript
  private setupSubscriptions(): void {
  ...}
 ```
 
 And make sure to add the dispose logic of that subscription into the `dispose` method of `BacklinksTreeDataProvider` as well.

Reply:
Gotcha makes sense